homeassistant/components/scrape/sensor.py

```diff
@@ -94,31 +41,10 @@ async def async_setup_platform(
     """Set up the Web scrape sensor."""
     coordinator: ScrapeCoordinator
     sensors_config: list[ConfigType]
-    if discovery_info is None:
-        async_create_issue(
-            hass,
-            DOMAIN,
-            "moved_yaml",
-            breaks_in_ha_version="2022.12.0",
-            is_fixable=False,
-            severity=IssueSeverity.WARNING,
-            translation_key="moved_yaml",
-        )
-        resource_config = vol.Schema(RESOURCE_SCHEMA, extra=vol.REMOVE_EXTRA)(config)
-        rest = create_rest_data_from_config(hass, resource_config)
-
-        scan_interval: timedelta = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
-        coordinator = ScrapeCoordinator(hass, rest, scan_interval)
-
-        sensors_config = [
-            vol.Schema(TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.ALLOW_EXTRA)(
-                config
-            )
-        ]
-
-    else:
-        coordinator = discovery_info["coordinator"]
-        sensors_config = discovery_info["configs"]
+    if TYPE_CHECKING:
+        assert discovery_info is not None
+    coordinator = discovery_info["coordinator"]
+    sensors_config = discovery_info["configs"]
```

Comment:
I think you should combine the type hint with the access - no need to a separate declaration anymore.

```suggestion    
    coordinator: ScrapeCoordinator = discovery_info["coordinator"]
    sensors_config: list[ConfigType] = discovery_info["configs"]
```